File: src/main/java/myLang/response/AbstractResponse.java
package myLang.response;

/**
 * Дополняет интерфейс {@link Response} реализуя хранение ответа и удобный конструктор.
 *
 * @param <T>
 */
abstract class AbstractResponse<T> implements Response<T> {
    final T response;

    /**
     * Конструктор ответа методов в интерфейсе {@link Response}.
     *
     * @param response тело ответа
     */
    AbstractResponse(T response) {
        this.response = response;
    }
}

File: src/main/kotlin/myLangKotlin/response/AbstractResponse.kt
package myLangKotlin.response

/**
 * Дополняет интерфейс [Response] реализуя хранение ответа и удобный конструктор.
 *
 * @param <T>
 */
abstract class AbstractResponse<T>
/**
 * Конструктор ответа методов в интерфейсе [Response].
 *
 * @param response тело ответа
 */
(override val response: T) : Response<T>
